File: DeepSpace/lib/supplies_package.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Deepspace
  class SuppliesPackage
    def initialize(ammoPower, fuelUnits, shieldPower)
      @ammoPower=ammoPower
      @fuelUnits=fuelUnits
      @shieldPower=shieldPower
    end
    
    def newCopy(s)
      SuppliesPackage.new(s.ammoPower, s.fuelUnits, s.shieldPower)
    end
    
    attr_reader :ammoPower
    attr_reader :fuelUnits
    attr_reader :shieldPower
    
  end
end

File: deepspace/src/deepspace/SuppliesPackage.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

/**
 *
 * @author antonio
 */
public class SuppliesPackage {
    
    private float ammoPower;
    private float fuelUnits;
    private float shieldPower;
    
    SuppliesPackage(float ammoPower, float fuelUnits, float shieldPower){
        this.ammoPower=ammoPower;
        this.fuelUnits=fuelUnits;
        this.shieldPower=shieldPower;
    }
    
    SuppliesPackage(SuppliesPackage s){
        this(s.getAmmoPower(), s.getFuelUnits(), s.getShieldPower());
    }

    public float getAmmoPower() {
        return ammoPower;
    }

    public float getFuelUnits() {
        return fuelUnits;
    }

    public float getShieldPower() {
        return shieldPower;
    }
    


}
